streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableOuterJoin.java

```diff
@@ -84,12 +89,28 @@ public void process(final K key, final Change<V1> change) {
             }
 
             R newValue = null;
+            final long resultTimestamp;
             R oldValue = null;
 
-            final V2 value2 = valueGetter.get(key);
-            if (value2 == null && change.newValue == null && change.oldValue == null) {
-                return;
+            final ValueAndTimestamp<V2> valueAndTimestamp2 = valueGetter.get(key);
+            final V2 value2 = getValueOrNull(valueAndTimestamp2);
+            final long timestamp2;
+            if (value2 == null) {
+                if (change.newValue == null && change.oldValue == null) {
+                    return;
+                }
+                timestamp2 = UNKNOWN;
+            } else {
+                timestamp2 = valueAndTimestamp2.timestamp();
+            }
+
+            final long timestamp1;
+            if (change.newValue == null) {
+                timestamp1 = UNKNOWN;
```

Comment:
Not sure if this is right... Even if the current value on the left is `null`, the update has come from somewhere, and I think it must have been triggered from the left, which means we actually got a `null` record with a known timestamp. Should we use this timestamp instead of `UNKNOWN` as the join result?

It seems weird to produce `-1` timestamps, since we can't handle them.

Reply:
Hmmm... That's a fair point. I don't think that we would ever propagate `-1` downstream (at least we don't see any in the tests... of course, tests might miss a case). However, the question is, what the result timestamp for the output should be. In the current PR, the result timestamp would be the timestamp from the right hand side (because it's an outer join -- `max(-1, RHS-ts)`). It might make sense to change it to the tombstone timestamp though. Thoughts?

Reply:
SGTM. It might be that we don't propagate it anyway, but it looks like you changed it to use the tombstone's timestamp, which seems right to me.

Reply:
Yes, I did update it. As I mentioned above (and thinking about it more), I agree that using the tombstone timestamp is better.